C/Linked-List/SinglyLinkedList.c

```diff
@@ -0,0 +1,335 @@
+        nptr->next = NULL;
+    }
+    else
+    {
+        nptr->next = aftptr->next;
+        aftptr->next = nptr;
+    }
+    printf("\nSUCCESSFULLY ADDED \n");
+};
+
+void delAtStart()
+{
+    struct node *ptr;
+    if (start == NULL)
+    {
+        printf("\nEmpty linked list(underflow)");
+        return;
+    }
+    ptr = start;
+    start = start->next;
+
+    printf("\nData deleted = %d\n", ptr->data);
+
+    /* Clears the memory occupied by first node*/
+    free(ptr);
+
+    printf("SUCCESSFULLY DELETED FIRST NODE FROM LIST\n");
+};
+void delAtEnd()
+{
+    struct node *ptr, *preptr;
+    if (start == NULL)
+    {
+        printf("\nEmpty linked list(underflow)");
+        return;
+    }
+    ptr = start;
+    while (ptr->next != NULL)
+    {
+        preptr = ptr;
+        ptr = ptr->next;
+    }
+    preptr->next = NULL;
+
+    printf("\nData deleted = %d\n", ptr->data);
+
+    /* Clears the memory occupied by first node*/
+    free(ptr);
+
+    printf("SUCCESSFULLY DELETED LAST NODE FROM LIST\n");
+}
+
+void delGivNode()
+{
+    struct node *ptr, *preptr;
+    int toDelete, key = 0;
+    if (start == NULL)
+    {
+        printf("\nEmpty linked list(underflow)");
+        return;
+    }
+    printf("\nEnter element to delete:- ");
+    scanf("%d", &toDelete);
+    ptr = start;
+    while (ptr->next != NULL)
+    {
+        if (ptr->data == toDelete)
+            break;
+        preptr = ptr;
+        key = 1;
+        ptr = ptr->next;
+    }
+    if (ptr == start)
+    {
+        printf("\nData deleted = %d\n", ptr->data);
+
+        /* Clears the memory occupied by first node*/
+        start = NULL;
+        printf("SUCCESSFULLY DELETED LAST NODE FROM LIST\n");
+        return;
+    }
+    if (key == 1)
+    {
+        preptr->next = ptr->next;
+        printf("\nData deleted = %d\n", ptr->data);
+
+        /* Clears the memory occupied by first node*/
+        free(ptr);
+
+        printf("SUCCESSFULLY DELETED LAST NODE FROM LIST\n");
+    }
+    else
+    {
+        printf("\nElement not found in LL");
+    }
+};
+
+int main()
+{
+    int ch;
+    char Opr[][31] = {"Create a List", "Traversal", "Searching in a list", "Insertion at the beginning", "Insertion at the end", "Insert before the given Node", "Insert after the given Node ", "Delete the first node", "Delete the last node", "Delete the given node", "Exit"};
+    do
+    {
+        printf("\n*****Main Menu*****\n");
+        for (int i = 0; i < sizeof(Opr) / 31; i++)
+            printf("%d. %s\n", i, Opr[i]);
+        printf("                        Enter your choice:- ");
+        scanf("%d", &ch);
+        switch (ch)
+        {
+        case 0:
+        {
+            int n;
+            printf("\nEnter no. of node :- ");
+            scanf("%d", &n);
+            createList(n);
+            break;
+        }
+        case 1:
+        {
+            traverse();
+            break;
+        }
+        case 2:
+        {
+            int s, skey = -1;
```

Comment:
In searching, kindly return the index position instead of memory address

Reply:
ok